core/src/main/java/io/parsingdata/metal/data/Slice.java

```diff
@@ -39,7 +40,7 @@ private Slice(final Source source, final BigInteger offset, final BigInteger len
     }
 
     public static Optional<Slice> createFromSource(final Source source, final BigInteger offset, final BigInteger length) {
-        if (offset.compareTo(ZERO) < 0 ||
+        if (checkNotNull(offset, "offset").compareTo(ZERO) < 0 ||
```

Comment:
Nice. I wonder why we didn't check this before?

Reply:
I think perhaps because these were protected or package-private constructors before so we more or less know whether it's necessary. Still, I prefer to be safe with such cheap checks.